src/util/misc.js

```diff
@@ -290,8 +290,147 @@
       );
     },
 
+    /**
+     * Returns the transform matrix according to `relationToObject`.\
+     * The returned matrix is relative to the entire object plane spread by a canvas,
+     * meaning it doesn't account for canvas viewport transform.
+     *
+     * `child` relation means `point` exists in the coordinate plane created by the object.
+     * In other words, point is measured acoording to object's center point
+     * meaning that if `point` is equal to (0,0) it is positioned at object's center.\
+     * `sibling` relation means `point` exists in the same coordinate plane as object.
+     * In other words they both relate to the same (0,0) and agree on every point.
+     *
+     * @static
+     * @memberOf fabric.util
+     * @param {fabric.Object} object
+     * @param {'sibling'|'child'} relationToObject
+     * @returns {number[]} plane matrix relative to the coordinate plane created by a canvas
+     */
+    getTransformMatrixByObject: function (object, relationToObject) {
+      if (relationToObject !== 'child' && relationToObject !== 'sibling') {
+        throw new Error('fabric.js: recieved bad argument ' + relationToObject);
+      }
+      return relationToObject === 'child' ?
+        object.calcTransformMatrix() :
+        object.group ?
+          object.group.calcTransformMatrix() :
+          fabric.iMatrix.concat();
+    },
```

Comment:
I m confused about this utility here.
I can't match the name with the code, maybe i just don't get what `matrixByObject` means.

Reply:
We pass an object and it's relation to the plane you what and it returns the matrix of the plane.
It can be either the plane that object creates (for it's children) or the plane it is contained in (where it is a sibling)

Reply:
or maybe fabric.Object.calcPlaneTransformMatrix() for the sibling/container, while for the child we just call calcTransformMatrix()

Reply:
sounds right to me

Reply:
I like this `fabric.Object.calcTransformMatrixByRelationship('sibling'|'child')`?